sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/templating/JavaParserTemplateProcessor.java

```diff
@@ -466,4 +462,61 @@ private void validateResponseStatus(BlockStmt body, HttpRequestContext method) {
         body.addStatement(StaticJavaParser.parseStatement("if (!expectedResponse) {"
             + " throw new RuntimeException(\"Unexpected response code: \" + responseCode); }"));
     }
+
+    private boolean addRequestBody(BlockStmt body, TypeMirror parameterType, String parameterName, boolean isJson,
+        Elements elementUtils, Types typeUtils) {
+
+        if (typeUtils.isSameType(parameterType,
+            elementUtils.getTypeElement("io.clientcore.core.models.binarydata.BinaryData").asType())) {
+            body.tryAddImportToParentCompilationUnit(BinaryData.class);
+            body.addStatement(
+                StaticJavaParser.parseStatement(String.format("BinaryData binaryData = %s;", parameterName)));
+            body.addStatement(StaticJavaParser.parseStatement("if (binaryData.getLength() != null) {"
+                + "httpRequest.getHeaders().set(HttpHeaderName.CONTENT_LENGTH, String.valueOf(binaryData.getLength()));"
+                + "httpRequest.setBody(binaryData); }"));
+        } else if (isJson) {
+            body.tryAddImportToParentCompilationUnit(SerializationFormat.class);
+            body.addStatement(StaticJavaParser
+                .parseStatement("httpRequest.setBody(BinaryData.fromObject(" + parameterName + ", jsonSerializer));"));
+        } else if (parameterType.getKind() == TypeKind.ARRAY) {
+            ArrayType arrayType = (ArrayType) parameterType;
+            if (arrayType.getComponentType().toString().equals("byte")) {
+                body.addStatement(StaticJavaParser
+                    .parseStatement(String.format("httpRequest.setBody(BinaryData.fromBytes(%s));", parameterName)));
+            }
+        } else if (typeUtils.isSameType(parameterType, elementUtils.getTypeElement("java.lang.String").asType())) {
+            body.addStatement(StaticJavaParser
+                .parseStatement(String.format("httpRequest.setBody(BinaryData.fromString(%s));", parameterName)));
+        } else if (typeUtils.isSameType(parameterType, elementUtils.getTypeElement("java.nio.ByteBuffer").asType())) {
+            body.tryAddImportToParentCompilationUnit(ByteBuffer.class);
+            body.addStatement(StaticJavaParser.parseStatement(
+                String.format("httpRequest.setBody(BinaryData.fromBytes(%s.array()));", parameterName)));
+        } else {
+            handleSerialization(body, parameterName);
+            return true;
```

Comment:
I don't fully understand the boolean returns going on here, why is this the only case that is returning `true` where it seems most other conditions are also adding a request body.

Reply:
Its a boolean for isSerializationFormatSet so that we dont initialize that value again in code.